rundeckapp/grails-app/controllers/rundeck/controllers/ProjectController.groovy

```diff
@@ -2425,7 +2453,33 @@ Authorization required: `configure` access for `project` resource type or `admin
             propValueBefore = new Properties([(key_): ''])
         }
 
-        def result=frameworkService.updateFrameworkProjectConfig(project.name,new Properties([(key_): value_]),null)
+        Properties projProp = new Properties([(key_): value_])
+
+        //validate plugin property values
+        def projectScopedConfigs = frameworkService.discoverScopedConfiguration(projProp, "project.plugin")
+        projectScopedConfigs.each { String svcName, Map<String, Map<String, String>> providers ->
+            final pluginDescriptions = pluginService.listPluginDescriptions(svcName)
+            providers.each { String provider, Map<String, String> providerConfig ->
+                def desc = pluginDescriptions.find { it.name == provider }
+                if (desc) {
+                    def validation = pluginService.validatePluginConfig(svcName, provider, providerConfig, null)
+                    if (!validation.valid) {
+                        Validator.Report report = validation.report
+                        errors << (
+                                report.errors ?
+                                        "${provider} configuration was invalid: " + report.errors :
+                                        "${provider} configuration was invalid"
+                        )
+                        return apiService.renderErrorFormat(response,[
+                                status: HttpServletResponse.SC_BAD_REQUEST,
+                                message:errors,
+                                format:respFormat
+                        ])
```

Comment:
the same problem here with returning within the closures, and rendering the error response